### PR TITLE
Add edit user profile

### PR DESCRIPTION
**Pull request by PerezIgnacio**
This pull request:
- Adds user profile edit menu
- Adds Feature tests for profile edit
- Adds Request tests for edit and update methods of RegistrationsController

Screen Shot 2019-05-16 at 12 15 26 PM

### CODE REVIEW

File: app/assets/javascripts/home.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+$(document).ready(function(){
+  $('.edit-link').on ('ajax:success', showEdit);
```
Comment (mdornel) [inline app/assets/javascripts/home.js]: remove space before parentheses

File: app/assets/javascripts/registrations.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+$(document).ready(function(){
+  $(document).on('click', '.edit-back', goBackToInfoMenu);
+  $(document).on('click', '#edit-user', editUserRequest);
+});
+
+function goBackToInfoMenu(e){
+  toggleInfoMenu()
```
Comment (mdornel) [inline app/assets/javascripts/registrations.js]: add ;

File: app/views/devise/registrations/_edit_form.html.haml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  .edit-title
+    %span.edit-back <
+    %span= t('views.registrations.edit.title')
+  .edit-form
+    = image_tag 'default-profile.png', srcset: { 'default-profile@2x.png' => '2x', 'default-profile@3x.png' => '3x'}, class: 'img-profile'
+    .info-name= current_user.name
+    .error-messages
+    .edit-form-fields
+      .edit-field
+        %label Email
```
Comment (mdornel) [inline app/views/devise/registrations/_edit_form.html.haml]: Use `I18n` for each string in the file

File: spec/features/user_edit_spec.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  given(:user) { create(:user, :confirmed) }
+
+  background do
+    login_as(user, scope: :user)
+    visit root_path
+    click_link 'Edit'
+    wait_for_ajax
+  end
+
+  scenario 'Shows user edit form', js: true do
```
Comment (mdornel) [inline spec/features/user_edit_spec.rb]: Adding `js: true` only in `feature 'Edit User'...`  at the top of the test should be enough. Don't need to add it in each scenario

File: spec/features/user_edit_spec.rb [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  context 'without information' do
+    scenario 'shows errors', js: true do
+      fill_form('', '', '', '')
+
+      expect(page).to have_content 'Email can\'t be blank'
+      expect(page).to have_content 'Current password can\'t be blank'
+    end
+  end
+
+  def fill_form(email, current_password, new_password, password_confirmation)
```
Comment (mdornel) [inline spec/features/user_edit_spec.rb]: What do you think about passing a hash of options as parameters? This way you will have only just one param and the inside of the method you can access to the attributes.

`options[:email]`
`options[:current_password]`

This approach could be better it the params keep growing

File: spec/rails_helper.rb [hunk 6]
```diff
@@ -24,6 +24,7 @@
 RSpec.configure do |config|
   config.include Devise::Test::ControllerHelpers, type: :controller
   config.include Devise::Test::IntegrationHelpers, type: :request
+  config.include WaitForAjax, type: :feature
```
Comment (mdornel) [inline spec/rails_helper.rb]: 👏 👏 Great stuff!

File: app/assets/javascripts/registrations.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+function toggleInfoMenu() {
+  $('.info-default').show();
+  $('.info-edit').remove();
+}
+
+function editUserRequest(e){
+  var url = $(e.target).parent().data('update-user-path');
```
Comment (jpascual1994) [inline app/assets/javascripts/registrations.js]: You can use `$(this)` instead of `$(e.target)`

File: app/assets/javascripts/registrations.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    url: url,
+    type: 'PUT',
+    data: params,
+  })
+  .done(function(data) {
+    toggleInfoMenu()
+  })
+  .fail(function(data) {
+    var errors = data.responseJSON;
+    $('.error-messages').empty();
```
Comment (jpascual1994) [inline app/assets/javascripts/registrations.js]: What about a variable for `$('.error-messages')`? It's just to not repeat it

File: app/assets/stylesheets/home.scss [hunk 9]
```diff
@@ -24,22 +27,25 @@ body { font-family: $open_sans_regular; }
...
   margin: 0px auto;
-  width: 98px;
 
   a { text-decoration: none; }
 }
 
-a:visited { color: black; }
+.edit-link { color: $default_blue_color; }
+
+.edit-link:visited { color: $default_blue_color; }
```
Comment (jpascual1994) [inline app/assets/stylesheets/home.scss]: You can nest it using `&`

```
.edit-link {
  color: $default_blue_color;
  
  &:visited { color: $default_blue_color; }
}
```

File: spec/features/user_edit_spec.rb (hunk 5 above)
Comment (PerezIgnacio) [reply spec/features/user_edit_spec.rb]: Great idea!

File: spec/features/user_edit_spec.rb [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+
+  background do
+    login_as(user, scope: :user)
+    visit root_path
+    click_link 'Edit'
+    wait_for_ajax
+  end
+
+  scenario 'Shows user edit form', js: true do
+    expect(page).to have_content 'EDIT PROFILE'
```
Comment (jpascual1994) [inline spec/features/user_edit_spec.rb]: You can use `I18n` on tests

**Pull request merged**